src/06-mocking-node-api/index.test.ts

```diff
@@ -1,21 +1,55 @@
-// Uncomment the code below and write your tests
-// import { readFileAsynchronously, doStuffByTimeout, doStuffByInterval } from '.';
+import { doStuffByInterval, doStuffByTimeout, readFileAsynchronously } from '.';
+import fs from 'fs';
+import fsp from 'fs/promises';
+import path from 'path';
+
+jest.mock('fs', () => ({
+  existsSync: jest.fn(),
+  readFile: jest.fn(),
+}));
+
+jest.mock('fs/promises', () => ({
+  readFile: jest.fn(),
+}));
+
+jest.mock('path', () => ({
+  join: jest.fn(),
+}));
 
 describe('doStuffByTimeout', () => {
   beforeAll(() => {
     jest.useFakeTimers();
+    jest.spyOn(global, 'setTimeout');
   });
 
   afterAll(() => {
     jest.useRealTimers();
   });
 
   test('should set timeout with provided callback and timeout', () => {
-    // Write your test here
+    const delay = 1000;
+    const callback = jest.fn();
+
+    doStuffByTimeout(callback, delay);
+
+    jest.advanceTimersByTime(delay);
+
+    expect(callback).toHaveBeenCalled();
+    expect(callback).toHaveBeenCalledTimes(1);
   });
 
   test('should call callback only after timeout', () => {
-    // Write your test here
+    const delay = 1000;
+    const callback = jest.fn();
+
+    doStuffByTimeout(callback, delay);
+
+    expect(callback).not.toHaveBeenCalled();
+
+    jest.runAllTimers();
```

Comment:
'only after timeout' mean that we need to emulate delay expire, exact how you do it in setInterval test.
this implementation not testing that stuff doing after delay, it testin that stuff run just after you run timers.